tests/test_config_entries.py

```diff
@@ -462,7 +462,15 @@ async def mock_unload_entry(
         assert result
         return result
 
-    mock_remove_entry = AsyncMock(return_value=None)
+    remove_entry_calls = []
+
+    async def mock_remove_entry(
+        hass: HomeAssistant, entry: config_entries.ConfigEntry
+    ) -> None:
+        """Mock removing an entry."""
+        # Check that the entry is not yet removed from config entries
+        assert hass.config_entries.async_get_entry(entry.entry_id)
```

Comment:
This is weird, in a follow-up we should adjust the implementation and this test so the entry is no longer in `hass.config_entries` when `async_remove_entry` is called.